github/commitstatus.go

```diff
@@ -31,12 +32,16 @@ func (e *StatusError) Error() string {
 	return fmt.Sprintf("%s\n%s", e.What, e.Details)
 }
 
-// API is the GitHub API endpoint.
+// GhDefaultHostname is the default GitHub hostname (used for git but not for the API)
+const GhDefaultHostname = "github.com"
+
+// API is the default GitHub API root URL.
 const API = "https://api.github.com"
```

Comment:
@marco-m-pix4d I know that I suggested removing this constant and just hardcode it in `ApiRoot`, but I tried and it make 2 tests a bit uglier so I decided to keep it.

Basically one would be required to configure the test server in 2 integration tests like:

```
Source{
  Server: github.ApiRoot(github.GhDefaultHostname)
}
```

compared to what we have now:
```
```
Source{
  Server: github.API
}
```

I don't see any gain with the change I proposed, so I kept it as it is.
```

Reply:
There is a gain: only one way to do it. It reduces confusion, since `github.API` doesn't exist, one is forced to use `github.ApiRoot(github.GhDefaultHostname)`, which is what we wanted.